### ISSUE TITLE
node_modules npm packages not being transpiled

### ISSUE THREAD

**Issue by GGAlanSmithee** [rollup/rollup-plugin-babel#4]
Hi!

According to the documentation, npm packages will be transpiled if not excluded

> All options are as per the Babel documentation, except options.include and options.exclude (each a minimatch pattern, or array of minimatch patterns), which determine which files are transpiled by Babel (by default, all files are transpiled).

I can't seem to get packages to transpile though. The [package](https://github.com/GGAlanSmithee/Entities) that is not being transpiled has a `jsnext:main` field which points to the entry of the non-bundled es6 code of that package.

**bundle.js**
```javascript
var rollup = require( 'rollup' );
var babel  = require('rollup-plugin-babel');
var npm  = require('rollup-plugin-npm');
var commonjs  = require('rollup-plugin-commonjs');

rollup.rollup({
  entry: 'main',
  plugins: [
    npm({
      jsnext: true,
      main: true
    }),
    commonjs({
      include: 'node_modules/**'
    }),
    babel({
      exclude: 'node_modules/three/**'
    })
  ]
}).then( function ( bundle ) {
  bundle.write({
    format: 'iife',
    dest: 'out.js'
  });
});
```

**.babelrc**
```javascript
{
    "presets": [ "es2015-rollup" ],
    "plugins": [ "transform-flow-strip-types", "syntax-flow" ]
}
```

**env**
`babel-core version 6.1.2`
`node v5.0.0`
`npm v3.3.11`
`rollup v0.20.5 with latest releases of plugins as of today`

**reproduce**
1. Start a new project
2. make files `bundle.js` like above
3. `npm i rollup rollup-plugin-babel rollup-plugin-npm rollup-plugin-commonjs gg-entities`
4. make file `main.js` and add `import { EntityManager } from 'gg-entities'` and use it in any way so that it gets included in the bundle, like `var em = new EntityManager();`
5. run `node bundle`
6. view `out.js`

**expected result**
code from npm packages should be transpiled

**actual result**
code from npm packages is not transpiled (code in the "targeted" src folder is though)

It is very likely that I have simply done something wrong, and if so sorry for reporting this as a bug. :)

Thanks for creating an awesome library!

**Comment by Rich-Harris** [comment]
Hmm. I think I experienced something similar the other day when I was trying out Babel 6 with Rollup itself – I think it might disregard your local `.babelrc` once you cross a package boundary. I found that I needed to pass the Babel configuration direct to the plugin...

```js
rollup.rollup({
  entry: 'main',
  plugins: [
    ...,
    babel({
      exclude: 'node_modules/three/**',
      presets: [ "es2015-rollup" ],
      plugins: [ "transform-flow-strip-types", "syntax-flow" ]
    })
  ]
}).then(...);
```

...rather than having it in `.babelrc` (you can't have both; the plugins get duplicated and the code gets horribly borked!), because it wouldn't transpile Acorn's source code otherwise. I eventually concluded that I was better off sticking with Babel 5 (i.e. `rollup-plugin-babel@1`) for the time being.

**Having said that**, it's probably better if the `jsnext:main` field of `gg-entities` points to pre-transpiled JavaScript (i.e. all language features *except* modules) – the rationale being that a consumer of your package shouldn't have to worry about your source code, only the code you're distributing. (This isn't a completely uncontroversial view – some people argue that we shouldn't transpile-on-publish, and should instead shift the burden to package consumers to decide what lengths they want to go to to support different environments, but personally I think that's clinically insane.) You could argue it's *slightly* different when you're consuming your own package, and that you want to avoid duplicating polyfills and helpers, but in my experience transpile-on-publish is the only way to avoid these sorts of nightmares.

**Comment by GGAlanSmithee** [comment]
@Rich-Harris thanks for you extensive answer and sorry I took this long to get back to you.

Firstly, there is definitely a good bit of logic to your reasoning regarding pre/post transpiling of distributed npm packages. I agree that it is unreasonable to put the burden of transpilation on the consumer, but I am looking at it more from an "all in or all out" point of view. Either you use the ES5 transpiled, umd-bundled, good-to-go package (`main`) or you use the raw, es6, unbundled import/export (`jsnext:main`) package. I will think more on this though.

To update this issue; my problems are solved! I did as you said and moved the `presets` / `plugins` declarations inline and it worked. Then I moved it back out to a .babelrc file, and it still worked - very wierd! I think it could have been some conflict between babel 6 and npm. I know there was some problems with babel and `node_modules` before npm v3, when a flat folder structure was introduced. Everything works fine now though.

BTW, I really want to thank you for authoring rollup and related packages, they are really great! For one example, it is the only bundler I have been able to use with Emscripten compiled packages. IT's virtually impossible to get webpack or browserify to accept packages that don't follow their idea of how a bundle should be packaged.

**Comment by Victorystick** [comment]
@GGAlanSmithee Glad to hear it! Let us know if you need any more help further on.

**Comment by trusktr** [comment]
I'm getting these sorts of silent errors with Rollup, no matter what `include`/`exclude` patterns I try. I've tried everything I can think of. I ended up switching to Webpack, it just works.

**Comment by trusktr** [comment]
Ahah!!!!! After switching to Webpack, I was able to use `webpack-rollup-loader` to get it to simply work, no more endless (and pointless) fiddling with `include`/`exclude` for Rollup plugins.

So my config looks like this: https://github.com/erikdesjardins/webpack-rollup-loader/issues/15#issue-288317827 (just mind that if you use Babel, you might get that error mentioned there, which is easy to fix by injecting a pointless line into the problematic file).

This to me proves that Rollup's resolution of node_modules is just not working. I literally tried everything I could, which spawned me to make [this PR](https://github.com/rollup/rollup-pluginutils/pull/24) which I didn't even actually test because it was too complicated, better just switching to Webpack.

**_Then I discovered that rollup-loader just works, using Webpack's node module resultion instead!_**
